src/main/java/com/yelp/nrtsearch/server/luceneserver/search/collectors/additional/TopHitsCollectorManager.java

```diff
@@ -149,6 +149,9 @@ public String getName() {
   @Override
   public void setSearchContext(SearchContext searchContext) {
     this.searchContext = searchContext;
+    // Backward compatibility: We reuse the highlightFetchTask from the SearchContext
+    this.retrievalContext.fetchTasks.setHighlightFetchTask(
+        searchContext.getFetchTasks().getHighlightFetchTask());
```

Comment:
It may be better to define highlighting as part of the collector definition, but that can be sorted out later